File: src/java/ArdsAdifApi/src/main/java/com/ards/adif/api/model/Region.java
package com.ards.adif.api.model;

import java.io.Serializable;
import javax.persistence.*;
import java.util.List;


/**
 * The persistent class for the region database table.
 * 
 */
@Entity
@Table(name="region")
@NamedQuery(name="Region.findAll", query="SELECT r FROM Region r")
public class Region implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@Column(unique=true, nullable=false)
	private Integer id;

	@Column(length=4)
	private String code;

	@Column(length=10)
	private String prefix;

	@Column(length=120)
	private String region;

	//bi-directional many-to-one association to Dxcc
	@ManyToOne
	@JoinColumn(name="dxcc_id")
	private Dxcc dxcc;

	//bi-directional many-to-one association to RegionApplicability
	@OneToMany(mappedBy="region")
	private List<RegionApplicability> regionApplicabilities;

	public Region() {
	}

	public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getCode() {
		return this.code;
	}

	public void setCode(String code) {
		this.code = code;
	}

	public String getPrefix() {
		return this.prefix;
	}

	public void setPrefix(String prefix) {
		this.prefix = prefix;
	}

	public String getRegion() {
		return this.region;
	}

	public void setRegion(String region) {
		this.region = region;
	}

	public Dxcc getDxcc() {
		return this.dxcc;
	}

	public void setDxcc(Dxcc dxcc) {
		this.dxcc = dxcc;
	}

	public List<RegionApplicability> getRegionApplicabilities() {
		return this.regionApplicabilities;
	}

	public void setRegionApplicabilities(List<RegionApplicability> regionApplicabilities) {
		this.regionApplicabilities = regionApplicabilities;
	}

	public RegionApplicability addRegionApplicability(RegionApplicability regionApplicability) {
		getRegionApplicabilities().add(regionApplicability);
		regionApplicability.setRegion(this);

		return regionApplicability;
	}

	public RegionApplicability removeRegionApplicability(RegionApplicability regionApplicability) {
		getRegionApplicabilities().remove(regionApplicability);
		regionApplicability.setRegion(null);

		return regionApplicability;
	}

}

File: src/dotnet/ArdsTools/Ards.Tools.Domain/Adif/ViewModels/Region.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ards.Tools.Domain.Adif.ViewModels
{
    [Table("region", Schema = "adif_view")]
    public partial class Region
    {
        [Key]
        [Display(Name = "RegionId")]
        [Column("region_id")]
        public int RegionId { get; set; }

        [Display(Name = "Code")]
        [Column("code")]
        [StringLength(4)]
        public string Code { get; set; }

        [Display(Name = "DXCC Code")]
        [Column("dxcc_code")]
        [StringLength(4)]
        public string DxccCode { get; set; }

        [Display(Name = "Region")]
        [Column("region")]
        [StringLength(120)]
        public string Region1 { get; set; }

        [Display(Name = "Prefix")]
        [Column("prefix")]
        [StringLength(10)]
        public string Prefix { get; set; }
    }
}
